### PR TITLE
Add dev container

### PR DESCRIPTION
**Pull request by jeromehardaway**
### Summary
🐳🛠️📦

This pull request adds support for developing the Vets-Who-Code app in a containerized environment using VS Code and Docker. It adds a `devcontainer.json` file and a `Dockerfile` to configure and build the development container.

> _`devcontainer.json`_
> _Configures the node image_
> _Spring of code begins_

### Walkthrough
*  Configure development container for the app ([link](https://github.com/Vets-Who-Code/vets-who-code-app/pull/519/files?diff=unified&w=0#diff-24ad71c8613ddcf6fd23818cb3bb477a1fb6d83af4550b0bad43099813088686R1-R22), [link](https://github.com/Vets-Who-Code/vets-who-code-app/pull/519/files?diff=unified&w=0#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R1-R11))

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      "settings": {
+        "terminal.integrated.shell.linux": "/bin/bash"
+      },
+      "extensions": [
+        "dbaeumer.vscode-eslint",
+        "ms-vscode.vscode-typescript-tslint-plugin"
+      ]
+    }
+  },
+  "forwardPorts": [3000],
```
Comment (tupleHunden) [inline .devcontainer/devcontainer.json:20]: Should this be a more unique port to avoid collisions with other processes?

Comment (tupleHunden) [reply .devcontainer/devcontainer.json:20]: I'm not super familiar with dev containers to know if this is self contained inside the container or would collide with other  processes outside it.

File: package-lock.json
Comment (tupleHunden) [inline package-lock.json:1]: Why is this here if Yarn is being used?  Shouldn't the container be using Yarn?

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (axecopfire) [reply .devcontainer/devcontainer.json:20]: This exposes running process on 3000, it's standard to directly expose it like this.

File: package-lock.json
Comment (axecopfire) [reply package-lock.json:1]: Looks like both containers in this PR (devcontainer and the Dockerfile) are using NPM. I don't actually know where the `"image": "vets-who-code:latest"` is coming from ([found here](https://github.com/Vets-Who-Code/vets-who-code-app/pull/519/files#diff-24ad71c8613ddcf6fd23818cb3bb477a1fb6d83af4550b0bad43099813088686R3)). It doesn't come up on search of `hub.docker.com`.

Comment (jeromehardaway) [reply package-lock.json:1]: I built the image.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,11 @@
+FROM node:18
+
+# Set working directory
+WORKDIR /src
+
+# Install dependencies
+COPY package*.json ./
+RUN npm install
```
Comment (axecopfire) [inline Dockerfile:8]: `npm install` runs in the `"postCreateCommand"` of the `devcontainer`, don't need to do it twice.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+# Add node_modules bin to PATH
+ENV PATH /app/node_modules/.bin:$PATH
```
Comment (axecopfire) [inline Dockerfile:11]: I don't see any reference to `/app`. I believe (though I may be wrong) since the working directory is `/src` you'll find the `node_modules` in there. Another note, it's a nice feature to add node_modules to path. But I've also felt comfortable just telling folks to `npx` modules.

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (jeromehardaway) [reply .devcontainer/devcontainer.json:20]: @tupleHunden Port 3000 is standard for this service and is isolated within the dev container, so it shouldn't collide with external processes.

File: .devcontainer/devcontainer.json [hunk 4]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "Vets-Who-Code",
+  "image": "vets-who-code:latest",
```
[the remark this answers is not in the document]
Comment (jeromehardaway) [reply .devcontainer/devcontainer.json:3]: @axecopfire The external image is intentional and has configurations that the Dockerfile doesn't cover. We're pulling vets-who-code:latest for that reason."

File: Dockerfile (hunk 2 above)
Comment (jeromehardaway) [reply Dockerfile:8]: @axecopfire Good point, I'll remove the RUN npm install from the Dockerfile since it's already being executed in the postCreateCommand of the devcontainer.json.

File: Dockerfile (hunk 3 above)
Comment (jeromehardaway) [reply Dockerfile]: @axecopfire You're correct; the working directory is /src, not /app. I'll update the PATH accordingly. As for adding node_modules to the PATH, it's to nice-to-have but not essential. We can use npx to execute local binaries if that's more straightforward.

File: package-lock.json
Comment (jeromehardaway) [reply package-lock.json:1]: @tupleHunden @axecopfire :

- No worries, we're set for both NPM and Yarn.
- "image": "vets-who-code:latest" is a custom image by me. Check it out on [Docker Hub](https://hub.docker.com/r/jhardaway/vets-who-code).

Comment (tupleHunden) [reply package-lock.json:1]: Why would the production env use `yarn`, but the development env use `npm`?  Can Docker and the dev container be updated to instead use what would be used in prod?

File: README.md [hunk 5]
```diff
@@ -66,6 +66,38 @@ $ npm run dev
...
+
+- [Docker](https://www.docker.com/products/docker-desktop)
+- [VS Code](https://code.visualstudio.com/)
+- [Remote - Containers extension for VS Code](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers)
+
+### Steps
+
+1. **Clone the repository**
+    ```sh
+    git clone [your-repository]
```
Comment (tupleHunden) [inline README.md:83]: This should be updated with the `git` repository instead of `[your-repository]`

Comment (jeromehardaway) [reply README.md]: Fixed

**Review by tupleHunden — APPROVED** [review]
